javascript/ql/src/Declarations/UnreachableMethodOverloads.ql

```diff
@@ -0,0 +1,131 @@
+/**
+ * @name Unreachable method overloads
+ * @description Having multiple overloads with the same parameter types in TypeScript
+ *              causes the latter overloads to become unreachable, as the TypeScript compiler
+ *              will always choose the textually first matching overload when selecting
+ *              which overload is called at a callsite.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/unreachable-method-overloads
+ * @precision high
+ * @tags correctness
+ *       typescript
+ */
+
+import javascript
+import semmle.javascript.Variables
+
+/**
+ * Gets the i'th parameter from the method signature.
```

Comment:
```suggestion
 * Gets the `i`th parameter from the method signature.
```

Reply:
Would this predicate make sense as a member predicate of `MethodSignature`?

Reply:
The member predicate should be on `MethodDeclaration`. 

Could be an idea. 
But in that case we should also create similar member predicates for at least `getCallSignature()` and `getTypeParameter()`. 
I lean towards not doing it, mostly because of the code duplication.